src/main/java/com/amalitech/gpuconfigurator/config/SecurityConfig.java

```diff
@@ -0,0 +1,31 @@
+package com.amalitech.gpuconfigurator.config;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationProvider;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
+
+
+@Configuration
+@EnableWebSecurity
+@RequiredArgsConstructor
+public class SecurityConfig {
+    private final AuthenticationProvider authenticationProvider;
+    private final JwtAuthenticationFilter jwtAuthenticationFilter;
+
+    @Bean
+    public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+        http.csrf(AbstractHttpConfigurer::disable)
+                    .authorizeHttpRequests(auth-> auth.requestMatchers("/v1/auth/**").permitAll().anyRequest().authenticated())
+                    .sessionManagement(session -> session.sessionCreationPolicy(SessionCreationPolicy.STATELESS))
```

Comment:
This way of chaining is not problematic or anything, but as a personal choice, I at times like to have it broken down into individual sections.

eg:
```
http.csrf(AbstractHttpConfigurer::disable)

http.authorizeHttpRequests(auth -> ...)

http.sessionManagement(session -> ...)

http.authenticationProvider(...)

return http.build()
```

But this is a personal choice anyway